kafka/src/main/java/brave/example/MessagingTracingFactory.java

```diff
@@ -0,0 +1,87 @@
+package brave.example;
+
+import brave.Tracing;
+import brave.baggage.BaggageField;
+import brave.baggage.BaggagePropagation;
+import brave.baggage.BaggagePropagationConfig;
+import brave.baggage.CorrelationScopeConfig;
+import brave.context.slf4j.MDCScopeDecorator;
+import brave.messaging.MessagingTracing;
+import brave.propagation.B3Propagation;
+import brave.propagation.CurrentTraceContext;
+import brave.propagation.CurrentTraceContext.ScopeDecorator;
+import brave.propagation.Propagation;
+import zipkin2.reporter.Sender;
+import zipkin2.reporter.brave.AsyncZipkinSpanHandler;
+import zipkin2.reporter.urlconnection.URLConnectionSender;
+
+import java.io.IOException;
+import java.util.logging.Logger;
+
+final class MessagingTracingFactory {
+  static final BaggageField USER_NAME = BaggageField.create("userName");
+
+  /** Decides how to name and tag spans. By default they are named the same as the http method. */
+  static MessagingTracing create(String serviceName) {
+    return MessagingTracing.create(tracing((System.getProperty("brave.localServiceName", serviceName))));
+  }
+
+  /** Controls aspects of tracing such as the service name that shows up in the UI */
+  static Tracing tracing(String serviceName) {
+    return Tracing.newBuilder()
+        .localServiceName(serviceName)
+        .supportsJoin(Boolean.parseBoolean(System.getProperty("brave.supportsJoin", "true")))
```

Comment:
Maybe I am missing something but does `supportJoin` works for messaging?

Reply:
Agree, it doesn't make sense in this scenario.